### DISCUSSION TITLE
How to use an alternate directory for config?

### DISCUSSION THREAD

**Discussion by jaraco** [tox-dev/tox#3017] [General]
In [mercurial/evolve#503](https://foss.heptapod.net/mercurial/evolve/-/merge_requests/503), I'm trying to add a `tox.ini` to the project, but that project would rather the file not be in the root of the directory and would prefer it to be instead in `contrib/tox.ini`.

When I move the tox.ini file, however, that means (a) it's not discovered without `-c contrib/tox.ini` and (b) assumptions about the locations of everything change (since `{tox_root}` is now `contrib/`).

I've found I can use `changedir = ../tests` in place of `changedir = tests`, and most things work like they did with `tox.ini` in the project root, except that the `{work_dir}` ends up being `contrib/.tox`, where I'd prefer it to be `./.tox`. I tried setting `work_dir = ../.tox`, but that creates `../.tox` for the environment creation, but that changes to `tests/../.tox` when running commands.

I tried setting `package_root` and `work_dir`, but had little success setting or referencing `{package_root}` (it never expanded).

Is there a straightforward way to reference a `tox.ini` but cause `{tox_root}` to be the current working directory even if `changedir` is used?

**Comment by gaborbernat** [comment]
You can use `-c contrib/tox.ini` combined with `--root .` to keep the tox root at the project root:

```bash
tox -c contrib/tox.ini --root .
```

The `--root` flag (or `tox_root` config) explicitly sets `{tox_root}` regardless of where the config file lives. This way `{work_dir}` defaults to `./.tox` and all paths resolve relative to the project root.
